src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+
+    async def _fetch_listings(self, criteria: Dict[str, Any]) -> List[Dict[str, Any]]:
+        """Fetch real estate listings from source"""
+        listings = []
+        page = 1
+        
+        while len(listings) < self.batch_size:
+            try:
+                params = self._build_params(criteria, page)
+                async with self.session.get(
+                    f"{self.base_url}/search",
+                    params=params
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)